src/GithubMergeTool/GithubMergeTool.cs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace GithubMergeTool
+{
+    public class GithubMergeTool
+    {
+        private static readonly Uri GithubBaseUri = new Uri("https://api.github.com/");
+
+        private readonly HttpClient _client;
+
+        public GithubMergeTool(
+            string username,
+            string password)
+        {
+            var client = new HttpClient();
+            client.BaseAddress = GithubBaseUri;
+
+            var authArray = Encoding.ASCII.GetBytes($"{username}:{password}");
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(
+                "Basic",
+                Convert.ToBase64String(authArray));
+            client.DefaultRequestHeaders.Add(
+                "user-agent",
+                "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.2;)");
+
+            _client = client;
+        }
+
+        /// <summary>
+        /// Create a merge PR.
+        /// </summary>
+        /// <returns>
+        /// (true, null) if the PR was created without error.
+        /// (false, null) if the PR wasn't created due to a PR already existing
+        /// or if the <paramref name="destBranch"/> contains all the commits
+        /// from <paramref name="srcBranch"/>.
+        /// (false, error response) if there was an error creating the PR.
+        /// </returns>
+        public async Task<(bool prCreated, HttpResponseMessage error)> CreateMergePr(
+            string repoOwner,
+            string repoName,
+            string srcBranch,
+            string destBranch)
+        {
+            string prTitle = $"Merge {srcBranch} to {destBranch}";
+
+            // Check to see if there's already a PR
+            // TODO: handle pagination
+            // This handles the 100 most recent PRs, which should work for now
+            HttpResponseMessage prsResponse = await _client.GetAsync(
+                $"repos/{repoOwner}/{repoName}/pulls?state=open&base={destBranch}&per_page=100");
```

Comment:
Can this also filter to ones being made by dotnet-bot? That might help address the pagination problem.

Reply:
I was just coming back to say this.

Reply:
Not a parameter in the `pulls` API.

Reply:
Wait, I think I misread something. Maybe.

Reply:
This works for me (only shows PR from me): https://api.github.com/repos/Microsoft/perfview/pulls?author=sharwell

Reply:
This doesn't appear to be documented: https://developer.github.com/v3/pulls/#list-pull-requests

What *is* documented is searching by both the author and branch name. Since we will now only have one merge PR for each two branches, I can remove the date from the branch name and make it predictable. I've done so in the latest commit.

Reply:
@agocke Just make sure that if somebody closes (but does not merge) the PR we are able to recover cleanly. I could imagine we screw up a merge so badly somebody just closes the PR and might not realize it's not working again.

Reply:
Good point. If someone merges the PR but forgets to delete the branch, that may block a new merge from being created. Would you be OK with the bot force-resetting the branch?